programs/clearing_house/src/lib.rs

```diff
@@ -1552,22 +1577,55 @@ pub mod clearing_house {
             )?;
 
             let user_position = &mut user.positions[position_index];
-            user_position.unsettled_pnl = user_position
-                .unsettled_pnl
-                .checked_sub(cast_to_i128(pnl_to_settle)?)
-                .ok_or_else(math_error!())?;
+            controller::position::update_unsettled_pnl(
+                user_position,
+                market,
+                -cast_to_i128(pnl_to_settle)?, // negative sign
+            )?;
         } else {
             // TODO handle socialized loss
-            let pnl_to_settle = user_unsettled_pnl.unsigned_abs();
+            let amm_pnl_pool_token_amount = cast_to_i128(get_token_amount(
+                market.amm.pnl_pool.balance(),
+                bank,
+                market.amm.pnl_pool.balance_type(),
+            )?)?;
+
+            // TODO parameterize?
+            let fraction_for_amm = 100; // up to 1% of user_unsettled_pnl for amm
+            let pnl_to_settle_to_amm = min(
+                cast_to_i128(market.amm.total_fee_minus_distributions)?
+                    .checked_sub(amm_pnl_pool_token_amount)
+                    .ok_or_else(math_error!())?,
+                user_unsettled_pnl
+                    .abs()
+                    .checked_div(fraction_for_amm)
+                    .ok_or_else(math_error!())?,
+            );
+
             update_bank_balances(
-                pnl_to_settle,
+                pnl_to_settle_to_amm.unsigned_abs(),
+                if pnl_to_settle_to_amm > 0 {
+                    &BankBalanceType::Deposit
```

Comment:
why shouldn't this always be a Deposit?

Reply:
not always, it will spend it if the pnl pool > total fee minus distribution